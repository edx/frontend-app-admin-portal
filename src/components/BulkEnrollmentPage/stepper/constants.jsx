export const STEPPER_TITLE = 'Learner enrollment';
export const ADD_LEARNERS_TITLE = 'Add learners';
export const REVIEW_TITLE = 'Review';
export const NEXT_BUTTON_TEXT = 'Next';
export const PREVIOUS_BUTTON_TEXT = 'Previous';
export const FINAL_BUTTON_TEXT = 'Enroll learners';
export const NEXT_BUTTON_TEST_ID = 'stepper-next';
export const PREV_BUTTON_TEST_ID = 'stepper-prev';
export const FINAL_BUTTON_TEST_ID = 'stepper-final';
