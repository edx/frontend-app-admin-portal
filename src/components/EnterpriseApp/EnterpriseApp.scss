@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/mixins";

.enterprise-app {
    position: relative;

    .content-wrapper {
        padding: 10px;

        @include media-breakpoint-up(lg) {
            & {
                padding-left: 51px;
            }
        }
    }
}
