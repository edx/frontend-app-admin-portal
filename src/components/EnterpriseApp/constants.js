/* eslint-disable import/prefer-default-export */

export const ROUTE_NAMES = {
  bulkEnrollment: 'catalog-management',
  subscriptionManagement: 'subscriptions',
};
