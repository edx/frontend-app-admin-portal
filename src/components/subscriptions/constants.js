export const TAB_ALL_USERS = 'TAB_ALL_USERS';
export const TAB_LICENSED_USERS = 'TAB_LICENSED_USERS';
export const TAB_PENDING_USERS = 'TAB_PENDING_USERS';
export const TAB_REVOKED_USERS = 'TAB_REVOKED_USERS';

export const PAGE_SIZE = 10;

// Subscription license statuses as defined on the backend
export const ACTIVATED = 'activated';
export const ASSIGNED = 'assigned';
export const REVOKED = 'revoked';
export const ALL_USERS = 'assigned,activated,revoked';

export const SUBSCRIPTIONS = 'Subscriptions';
export const SUBSCRIPTION_OVERVIEW = 'Subscription Overview';
export const SUBSCRIPTION_USERS = 'Subscription Users';
export const SUBSCRIPTION_USERS_OVERVIEW = 'Subscription Users Overview';

export const NETWORK_ERROR_MESSAGE = 'Error occurred while loading the data.';
export const DEFAULT_PAGE = 1;

// used to determine whether to show the revocation cap messaging
export const SHOW_REVOCATION_CAP_PERCENT = 10;
