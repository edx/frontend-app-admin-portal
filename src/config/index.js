import qs from 'query-string';

const hasFeatureFlagEnabled = (featureFlag) => {
  const { features } = qs.parse(window.location.search);
  return features && features.split(',').includes(featureFlag);
};

const configuration = {
  BASE_URL: process.env.BASE_URL,
  LMS_BASE_URL: process.env.LMS_BASE_URL,
  LOGIN_URL: process.env.LOGIN_URL,
  LOGOUT_URL: process.env.LOGOUT_URL,
  REFRESH_ACCESS_TOKEN_ENDPOINT: process.env.REFRESH_ACCESS_TOKEN_ENDPOINT,
  DATA_API_BASE_URL: process.env.DATA_API_BASE_URL,
  ECOMMERCE_API_BASE_URL: process.env.ECOMMERCE_API_BASE_URL,
  SECURE_COOKIES: process.env.NODE_ENV !== 'development',
  SEGMENT_KEY: process.env.SEGMENT_KEY,
  ACCESS_TOKEN_COOKIE_NAME: process.env.ACCESS_TOKEN_COOKIE_NAME,
  CSRF_COOKIE_NAME: process.env.CSRF_COOKIE_NAME,
};

const features = {
  CODE_MANAGEMENT: process.env.FEATURE_FLAGS.CODE_MANAGEMENT || hasFeatureFlagEnabled('CODE_MANAGEMENT'),
};

export { configuration, features };
