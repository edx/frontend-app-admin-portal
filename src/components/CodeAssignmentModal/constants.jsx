import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
import { MODAL_TYPES } from '../EmailTemplateForm/constants';
import { EMAIL_TEMPLATE_FIELDS } from '../EmailTemplateForm';
import CheckboxWithTooltip from '../ReduxFormCheckbox/CheckboxWithTooltip';

export const ASSIGNMENT_ERROR_TITLES = {
  [MODAL_TYPES.assign]: 'Unable to assign codes',
  [MODAL_TYPES.save]: 'Unable to save template',
};
export const EMAIL_TEMPLATE_NUDGE_EMAIL_ID = 'enable-nudge-emails';

export const ASSIGNMENT_MODAL_FIELDS = {
  ...EMAIL_TEMPLATE_FIELDS,
  [EMAIL_TEMPLATE_NUDGE_EMAIL_ID]: {
    name: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
    id: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
    component: CheckboxWithTooltip,
    className: 'auto-reminder-wrapper',
    icon: faInfoCircle,
    altText: 'More information',
    tooltipText: 'edX will remind learners to redeem their code 3, 10, and 19 days after you assign it.',
    label: 'Automate reminders',
  },
};
