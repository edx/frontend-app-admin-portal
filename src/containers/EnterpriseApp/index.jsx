import { connect } from 'react-redux';

import EnterpriseApp from '../../components/EnterpriseApp';

import { fetchPortalConfiguration } from '../../data/actions/portalConfiguration';
import { toggleSidebarToggle } from '../../data/actions/sidebar';

const mapStateToProps = (state) => {
  const enterpriseListState = state.table['enterprise-list'] || {};

  return {
    enterprises: enterpriseListState.data,
    error: state.portalConfiguration.error,
  };
};

const mapDispatchToProps = dispatch => ({
  fetchPortalConfiguration: (slug) => {
    dispatch(fetchPortalConfiguration(slug));
  },
  toggleSidebarToggle: () => {
    dispatch(toggleSidebarToggle());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(EnterpriseApp);
