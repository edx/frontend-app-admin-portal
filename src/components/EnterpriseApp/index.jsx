import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import CoursewarePage from '../../containers/CoursewarePage';
import AdminPage from '../../containers/AdminPage';
import CodeManagementPage from '../../containers/CodeManagementPage';
import RequestCodesPage from '../../containers/RequestCodesPage';
import SupportPage from '../SupportPage';
import NotFoundPage from '../NotFoundPage';
import ErrorPage from '../ErrorPage';

import { removeTrailingSlash } from '../../utils';
import { features } from '../../config';

class EnterpriseApp extends React.Component {
  renderError(error) {
    return (
      <ErrorPage
        status={error.response && error.response.status}
        message={error.message}
      />);
  }

  render() {
    const { error, match } = this.props;
    const baseUrl = match.url;
    const { enterpriseSlug } = match.params;

    if (error) {
      return this.renderError(error);
    }

    return (
      <React.Fragment>
        <Switch>
          <Redirect
            exact
            from={baseUrl}
            to={`${removeTrailingSlash(baseUrl)}/admin/learners`}
          />
          <Route exact path={`${baseUrl}/courses/:courseId`} component={CoursewarePage} />
          <Route
            exact
            path={`${baseUrl}/admin/learners/:actionSlug?`}
            render={routeProps => <AdminPage {...routeProps} enterpriseSlug={enterpriseSlug} />}
          />
          {features.CODE_MANAGEMENT &&
            <React.Fragment>
              <Route
                exact
                path={`${baseUrl}/admin/codes`}
                render={routeProps =>
                  <CodeManagementPage {...routeProps} enterpriseSlug={enterpriseSlug} />
                }
              />
              <Route
                exact
                path={`${baseUrl}/admin/codes/request`}
                render={routeProps =>
                  <RequestCodesPage {...routeProps} enterpriseSlug={enterpriseSlug} />
                }
              />
            </React.Fragment>
          }
          <Route exact path={`${baseUrl}/support`} component={SupportPage} />
          <Route path="" component={NotFoundPage} />
        </Switch>
      </React.Fragment>
    );
  }
}

EnterpriseApp.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string.isRequired,
    params: PropTypes.shape({
      enterpriseSlug: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  error: PropTypes.instanceOf(Error),
};

EnterpriseApp.defaultProps = {
  error: null,
};

export default EnterpriseApp;
