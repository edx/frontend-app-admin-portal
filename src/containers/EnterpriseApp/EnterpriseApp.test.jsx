import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { breakpoints } from '@edx/paragon';

import EnterpriseApp from './index';

import { TOGGLE_SIDEBAR_TOGGLE } from '../../data/constants/sidebar';

import { features } from '../../config';

features.CODE_MANAGEMENT = true;

const mockStore = configureMockStore([thunk]);
const initialState = {
  authentication: {
    roles: ['enterprise_admin:*'],
  },
  userAccount: {
    loaded: true,
    isActive: true,
  },
  dashboardAnalytics: {},
  portalConfiguration: {
    enterpriseId: 'test-enterprise-id',
    enableCodeManagementScreen: true,
    enableSubscriptionManagementScreen: true,
    enableAnalyticsScreen: true,
  },
  csv: {},
  table: {
    'enterprise-list': {},
  },
  sidebar: {
    isExpanded: false,
    isExpandedByToggle: false,
  },
};

const EnterpriseAppWrapper = ({ store, ...props }) => (
  <MemoryRouter initialEntries={['/test-enterprise-slug/admin/learners']}>
    <Provider store={store}>
      <EnterpriseApp
        match={{
          url: '/test-enterprise-slug',
          params: {
            enterpriseSlug: 'test-enterprise-slug',
          },
        }}
        location={{
          pathname: '/test-enterprise-slug/admin/learners',
        }}
        history={{
          replace: () => {},
        }}
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

EnterpriseAppWrapper.defaultProps = {
  store: mockStore({ ...initialState }),
};

EnterpriseAppWrapper.propTypes = {
  store: PropTypes.shape({}),
};

describe('EnterpriseApp', () => {
  it('renders not found page correctly', () => {
    const tree = renderer
      .create((
        <EnterpriseAppWrapper
          match={{
            url: '/',
            params: {
              enterpriseSlug: '',
            },
          }}
        />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders error page correctly', () => {
    const store = mockStore({
      ...initialState,
      portalConfiguration: {
        ...initialState.portalConfiguration,
        error: Error('test error'),
      },
    });

    const tree = renderer
      .create((
        <EnterpriseAppWrapper store={store} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles location change properly', () => {
    // There is some logic where we collapse the sidebar on menu click on mobile
    // so we test that here as well. Note that we need to set the window width
    // to mobile first.
    const initialWidth = global.innerWidth;
    global.innerWidth = breakpoints.small.minWidth;

    const store = mockStore({
      ...initialState,
      sidebar: {
        ...initialState.sidebar,
        isExpandedByToggle: true,
      },
    });

    const wrapper = mount(
      <EnterpriseAppWrapper store={store} />,
      { attachTo: document.body },
    );

    wrapper.setProps({
      location: {
        pathname: '/test-enterprise-slug/admin/codes',
      },
    });

    // ensure focus is set on content wrapper
    expect(document.activeElement.className).toEqual('content-wrapper');
    global.innerWidth = initialWidth;
  });

  it('toggles sidebar toggle on componentWillUnmount', () => {
    const store = mockStore({
      ...initialState,
      sidebar: {
        ...initialState.sidebar,
        isExpandedByToggle: true,
      },
    });

    const wrapper = mount((
      <EnterpriseAppWrapper
        store={store}
      />
    ));

    // clear existing actions
    store.clearActions();

    // unmount component to trigger componentWillUnmount lifecycle method
    wrapper.unmount();

    // ensure the TOGGLE_SIDEBAR_TOGGLE action is dispatched
    const actions = store.getActions().filter(action => action.type === TOGGLE_SIDEBAR_TOGGLE);
    expect(actions).toHaveLength(1);
  });
});
