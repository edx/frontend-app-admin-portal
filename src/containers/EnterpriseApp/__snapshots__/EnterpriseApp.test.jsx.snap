// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterpriseApp renders error page correctly 1`] = `
<main
  role="main"
>
  <div
    className="container-fluid"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col"
      >
        <h1>
          Error
        </h1>
        <div
          className="alert fade alert-danger show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className=""
            >
              <div
                className="message"
              >
                test error
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`EnterpriseApp renders not found page correctly 1`] = `
<div
  className="enterprise-app"
>
  <nav
    aria-label="sidebar"
    className="sidebar border-right h-100 d-none d-lg-flex has-shadow"
    id="sidebar"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="sidebar-content py-2"
    >
      <ul
        className="nav nav-pills flex-column m-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link text-left rounded-0"
            href="//admin/learners"
            onClick={[Function]}
          >
            <span
              aria-hidden={true}
              className="fa fa-line-chart"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Learner Report
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link text-left rounded-0"
            href="//admin/coupons"
            onClick={[Function]}
          >
            <span
              aria-hidden={true}
              className="fa fa-tags"
              id="Icon2"
            />
            <span
              className="sr-only"
            >
              Code Management
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link text-left rounded-0"
            href="//admin/subscriptions"
            onClick={[Function]}
          >
            <span
              aria-hidden={true}
              className="fa fa-credit-card"
              id="Icon3"
            />
            <span
              className="sr-only"
            >
              Subscription Management
            </span>
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="content-wrapper"
    style={
      Object {
        "paddingLeft": 10,
      }
    }
    tabIndex="-1"
  >
    <main
      role="main"
    >
      <div
        className="container-fluid mt-3"
      >
        <div
          className="text-center py-5"
        >
          <h1>
            404
          </h1>
          <p
            className="lead"
          >
            Oops, sorry we can't find that page!
          </p>
          <p>
            Either something went wrong or the page doesn't exist anymore.
          </p>
        </div>
      </div>
    </main>
  </div>
</div>
`;
